iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -163,6 +167,26 @@ public void commitTask(TaskAttemptContext originalContext) throws IOException {
               LOG.info("CommitTask found no serialized table in config for table: {}.", output);
             }
           }, IOException.class);
+
+      // Merge task has merged several files into one. Hence we need to remove the stale files.
```

Comment:
could we extract this part into some cleaner method?

Reply:
Done.